src/array.jl

```diff
@@ -22,8 +25,11 @@ function unsafe_free!(xs::CuArray)
   return
 end
 
-Base.cconvert(::Type{Ptr{T}}, x::CuArray{T}) where T = x.buf
-Base.cconvert(::Type{Ptr{Void}}, x::CuArray) = x.buf
+unsafe_buffer(xs::CuArray) =
+  Mem.Buffer(xs.buf.ptr+xs.offset, sizeof(xs), xs.buf.ctx)
```

Comment:
How do you make sure `offset+sizeof(xs)` never exceeds the original buffer, and goes out of bounds?
You could also use `Mem.view(::Buffer, offset)` for this, which does a bounds check on the new buffer.

Reply:
If that was the case then the original CuArray object would have been invalid to begin with. None of our constructors should allow this to happen.

Reply:
Although I think I still need to put in a bounds check in one place